test/framework/src/main/java/org/elasticsearch/bootstrap/BootstrapForTesting.java

```diff
@@ -116,10 +118,27 @@
                     // in case we get fancy and use the -integration goals later:
                     perms.add(new FilePermission(coverageDir.resolve("jacoco-it.exec").toString(), "read,write"));
                 }
-                // intellij hack: intellij test runner wants setIO and will
-                // screw up all test logging without it!
+
+                Set<URL> codebases = JarHell.parseClassPath();
+                Map<String, URL> shortNameToCodebases = Security.resolveShortNames(
+                        codebases.stream().filter(url -> url.getPath().endsWith(".jar")).collect(toList()));
                 if (System.getProperty("tests.gradle") == null) {
+                    // intellij hack: intellij test runner wants setIO and will
+                    // screw up all test logging without it!
                     perms.add(new RuntimePermission("setIO"));
+
+                    /* If we are running in Eclipse then assign the build-eclipse directory
+                     * the permissions of the we would assign to the built jar.*/
+                    Path client = PathUtils.get("elasticsearch", "client", "rest", "src", "main", "build-eclipse", "1");
+                    for (URL url : codebases) {
+                        if (PathUtils.get(url.toURI()).endsWith(client)) {
+                            URL previous = shortNameToCodebases.put("elasticsearch-rest-client-6.0.0-beta1-SNAPSHOT-nodeps.jar", url);
```

Comment:
This line will require changes every time we bump versions. I have an idea for how to remove the need for that (making the policy files templated and processed by gradle, with the current version passed in). I think we should be moving away from hardcoded refs to version, not adding more.